server/graphql/v2/object/Host.js

```diff
@@ -437,6 +452,124 @@ export const Host = new GraphQLObjectType({
           };
         },
       },
+      contributionStats: {
+        type: new GraphQLNonNull(ContributionStats),
+        args: {
+          account: {
+            type: new GraphQLList(new GraphQLNonNull(AccountReferenceInput)),
+            description: 'A collection of accounts for which the contribution stats should be returned.',
+          },
+          dateFrom: {
+            type: GraphQLDateTime,
+            description: 'Calculate contribution statistics beginning from this date.',
+          },
+          dateTo: {
+            type: GraphQLDateTime,
+            description: 'Calculate contribution statistics until this date.',
+          },
+        },
+        async resolve(host, args, req) {
+          if (!req.remoteUser?.isAdmin(host.id)) {
+            throw new Unauthorized('You need to be logged in as an admin of the host to see the contribution stats.');
+          }
+          const where = { HostCollectiveId: host.id, kind: 'CONTRIBUTION' };
+          const dateRange = getFilterDateRange(args.dateFrom, args.dateTo);
+          if (dateRange) {
+            where.createdAt = dateRange;
+          }
+          if (args.account) {
+            const collectiveIds = args.account.map(async account => {
+              const accountObj = await fetchAccountWithReference(account, { throwIfMissing: true });
+              return accountObj.id;
+            });
+            await Promise.all(collectiveIds).then(collectiveIds => {
+              where.CollectiveId = { [Op.in]: collectiveIds };
+            });
+          }
+          const contributionsCount = await models.Transaction.count({
+            where,
+          });
+          const contributionsAmountSum = await models.Transaction.sum('amount', where);
+          const dailyAverageIncomeAmount = contributionsCount !== 0 ? contributionsAmountSum / contributionsCount : 0;
+          return {
+            contributionsCount,
+            oneTimeContributionsCount: models.Transaction.count({
+              where,
+              include: [{ model: models.Order, where: { interval: null } }],
+            }),
+            recurringContributionsCount: models.Transaction.count({
+              where,
+              include: [{ model: models.Order, where: { interval: { [Op.ne]: null }, status: 'ACTIVE' } }],
+            }),
+            dailyAverageIncomeAmount: {
+              valueInCents: dailyAverageIncomeAmount,
+              value: dailyAverageIncomeAmount / 100,
+              currency: host.currency,
```

Comment:
From `server/graphql/v2/object/Amount.js`, this object only expects a `value` (in cents) and a `currency`. The properties themselves are generated by loaders. Here, all the amounts returned are divided by 100.

```suggestion
              value: dailyAverageIncomeAmount,
              currency: host.currency,
```

Reply:
Thanks for pointing it out. I've corrected this. 👍🏽